packit_service/worker/handlers/centosmsg_handlers.py

```diff
@@ -0,0 +1,144 @@
+import logging
+from typing import Optional, Callable, Union
+
+from ogr.abstract import GitProject
+
+from packit.config import JobConfig, PackageConfig, JobType
+from packit_service.config import (
+    ServiceConfig,
+    PagurePackageConfigGetter,
+)
+from packit_service.service.events import (
+    PushGitHubEvent,
+    TheJobTriggerType,
+    PullRequestPagureEvent,
+    PushPagureEvent,
+)
+from packit_service.worker.build import CoprBuildJobHelper
+from packit_service.worker.handlers.abstract import use_for, JobHandler
+from packit_service.worker.result import HandlerResults
+
+logger = logging.getLogger(__name__)
+
+
+class AbstractPagureJobHandler(JobHandler, PagurePackageConfigGetter):
+    pass
+
+
+class AbstractPagureCoprBuildHandler(AbstractPagureJobHandler):
+    event: Union[PullRequestPagureEvent]
+
+    def __init__(
+        self,
+        config: ServiceConfig,
+        job_config: JobConfig,
+        event: Union[PullRequestPagureEvent, PushPagureEvent],
+    ):
+        super().__init__(config=config, job_config=job_config, event=event)
+
+        # lazy property
+        self._copr_build_helper: Optional[CoprBuildJobHelper] = None
+        self._package_config: Optional[PackageConfig] = None
+        self._project: Optional[GitProject] = None
+
+    @property
+    def copr_build_helper(self) -> CoprBuildJobHelper:
+        if not self._copr_build_helper:
+            self._copr_build_helper = CoprBuildJobHelper(
+                config=self.config,
+                package_config=self.package_config,
+                project=self.project,
+                event=self.event,
+                job=self.job_config,
+            )
+        return self._copr_build_helper
+
+    @property
+    def project(self) -> GitProject:
+        if not self._project:
+            self._project = self.event.get_project()
+        return self._project
+
+    @property
+    def package_config(self) -> PackageConfig:
+        if not self._package_config:
+            self._package_config = self.event.get_package_config()
+            self._package_config.upstream_project_url = self.event.project_url
+        return self._package_config
+
+    def run(self) -> HandlerResults:
+        return self.copr_build_helper.run_copr_build()
+
+    def pre_check(self) -> bool:
+        is_copr_build: Callable[
+            [JobConfig], bool
+        ] = lambda job: job.type == JobType.copr_build
+
+        if self.job_config.type == JobType.tests and any(
+            filter(is_copr_build, self.package_config.jobs)
+        ):
+            logger.info(
+                "Skipping build for testing. The COPR build is defined in the config."
+            )
+            return False
+        return True
+
+
+@use_for(job_type=JobType.copr_build)
+@use_for(job_type=JobType.build)
+class PagurePullRequestCoprBuildHandler(AbstractPagureCoprBuildHandler):
```

Comment:
Is this class intended to be responsible for starting builds when a pull-request is filed/updated in CentOS infra? If yes, could we call it simply `PullRequestHandler`? This name is too long, and it has too many details that can be interfered from the context or is irrelevant at this level of abstraction.

Reply:
I have followed current architecture. 

There are different implementation how we are getting configs for GitHub and Pagure because of different api capabilities. 
Event.get_package_config() and AbstractGithubJobHandler/AbstractPagureJobHandler (via GithubPackageConfigGetter/PagurePackageConfigGetteris) GitService type dependent, thus preventing of using more general class. I think this breaking of SRP, i will raise arch day topic for this.